projects/kit/directives/data-list-dropdown-manager/data-list-dropdown-manager.directive.ts

```diff
@@ -36,6 +41,10 @@ export class TuiDataListDropdownManagerDirective implements AfterViewInit {
     @ContentChildren(TuiDropdownDirective, {read: ElementRef, descendants: true})
     private readonly elements: QueryList<ElementRef<HTMLElement>> = EMPTY_QUERY;
 
+    constructor(
+        @Self() @Inject(TuiDestroyService) private readonly destroy$: TuiDestroyService,
+    ) {}
+
     ngAfterViewInit(): void {
         this.right$.subscribe(index => {
```

Comment:
You should add it here too. There was no `takeUntil` in this component, because it comes from `@QueryList` which completes after component is destroyed, but we did not consider `shareReplay`.

Reply:
done ✔️